diff --git a/setup.py b/setup.py
index 930e689..fc95539 100644
--- a/setup.py
+++ b/setup.py
@@ -181,7 +181,7 @@ with open('./requirements.txt') as f:
 setup(
     name='openPMD-api',
     # note PEP-440 syntax: x.y.zaN but x.y.z.devN
-    version='0.17.1',
+    version='0.17.1.post1',
     author='Axel Huebl, Franz Poeschel, Fabian Koller, Junmin Gu',
     author_email='axelhuebl@lbl.gov, f.poeschel@hzdr.de',
     maintainer='Axel Huebl',
